### ISSUE TITLE
 file-5.25-r0 do_fetch: Fetcher failure: Unable to find revision in branch master

### ISSUE THREAD

**Issue by javiroman** [comodoo/meta-comodoo#28]
In branch krogoth the package "file" fails doing the fetching tasks:

```
bitbake -DDD file -c do_fetch
```

```
DEBUG: file-5.25-r0 do_fetch: Running export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export PATH="/home/javiroman/HACK/yocto/poky/build/tmp/sysroots-uninative/x86_64-linux/usr/bin:/home/javiroman/HACK/yocto/poky/scripts:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/x86_64-linux/usr/bin/arm-poky-linux-gnueabi:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/raspberrypi3/usr/bin/crossscripts:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/x86_64-linux/usr/sbin:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/x86_64-linux/usr/bin:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/x86_64-linux/sbin:/home/javiroman/HACK/yocto/poky/build/tmp/sysroots/x86_64-linux/bin:/home/javiroman/HACK/yocto/poky/scripts:/home/javiroman/HACK/yocto/poky/bitbake/bin:/usr/lib64/qt-3.3/bin:/usr/lib64/ccache:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/javiroman/.local/bin:/home/javiroman/bin"; export HOME="/home/javiroman"; git -c core.fsyncobjectfiles=0 branch --contains 789cfc7d727cee1c7cfb7d29c09162e2399285c5 --list master 2> /dev/null | wc -l
DEBUG: file-5.25-r0 do_fetch: Mirror fetch failure for url http://sources.openembedded.org/git2_github.com.file.file.git.tar.gz (original url: git://github.com/file/file.git)
DEBUG: file-5.25-r0 do_fetch: Fetcher failure: Unable to find revision 789cfc7d727cee1c7cfb7d29c09162e2399285c5 in branch master even from upstream
ERROR: file-5.25-r0 do_fetch: Fetcher failure: Unable to find revision 789cfc7d727cee1c7cfb7d29c09162e2399285c5 in branch master even from upstream
DEBUG: file-5.25-r0 do_fetch: Python function base_do_fetch finished
DEBUG: file-5.25-r0 do_fetch: Python function do_fetch finished
ERROR: file-5.25-r0 do_fetch: Function failed: Fetcher failure for URL: 'git://github.com/file/file.git'. Unable to fetch URL from any source.
ERROR: Logfile of failure stored in: /home/javiroman/HACK/yocto/poky/build/tmp/work/cortexa7hf-neon-vfpv4-poky-linux-gnueabi/file/5.25-r0/temp/log.do_fetch.3156
ERROR: Task 0 (/home/javiroman/HACK/yocto/poky/meta/recipes-devtools/file/file_5.25.bb, do_fetch) failed with exit code '1'
DEBUG: Teardown for bitbake-worker
NOTE: Tasks Summary: Attempted 1 tasks of which 0 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/javiroman/HACK/yocto/poky/meta/recipes-devtools/file/file_5.25.bb, do_fetch
Summary: There were 2 WARNING messages shown.
Summary: There were 2 ERROR messages shown, returning a non-zero exit code.
```

**Comment by javiroman** [comment]
Trying to reproduce manually:

```
$ wget http://downloads.yoctoproject.org/mirror/sources/git2_github.com.file.file.git.tar.gz
```
The command to check the git commit targeted:

```
$ git -c core.fsyncobjectfiles=0 branch --contains 789cfc7d727cee1c7cfb7d29c09162e2399285c5 --list master 2> /dev/null | wc -l
0
```
However:
```
git -c core.fsyncobjectfiles=0 branch --contains 789cfc7d727cee1c7cfb7d29c09162e2399285c5 
  badmaster
  origin
```

The downloaded tar.gz only contains the branches "badmaster" and "origin", both of them contains the targeted commit. So the original tar.gz from Yocto download is not correctly packed.

**Comment by javiroman** [comment]
In order to fix this issue, we can copy the file package to our comodoo-meta layer and override the original package. Any other ideas?

**Comment by indisoluble** [comment]
I had this same problem while a was working on issue #25. Let me explain:
- Originally, I indicated that `poky` had to be reset to commit `8c69f7d56cbd496aa01ba0738675a170826a536b`. In this commit, the recipe `meta/recipes-devtools/file/file_5.25.bb` pointed to `f45db89ddc91692b662fffbabbdafc7bc4c00f5e`.
- However, in the new **README** I indicate that `poky` has to be reset to `b4df9df4624a300923a5dfc5a8f7157bef145a2a`. In this commit, the recipe `meta/recipes-devtools/file/file_5.25.bb` points to `789cfc7d727cee1c7cfb7d29c09162e2399285c5`.

Now, I had to specify a new commit for `poky` because the commit `f45db89ddc91692b662fffbabbdafc7bc4c00f5e` disappeared. I guess we have to do the same now because the commit `789cfc7d727cee1c7cfb7d29c09162e2399285c5` has also disappeared. I.e. I have to update again **README** to point `poky` to a newer commit.
